qa/rpc-tests/feature_fedpeg.py

```diff
@@ -240,25 +240,50 @@ def run_test(self):
         if sidechain.gettransaction(pegtxid1)["confirmations"] != 6:
             raise Exception("Peg-in should be back to 6 confirms.")
 
-        # Do many claims in mempool
-        n_claims = 5
+        # Do multiple claims in mempool
+        n_claims = 6
 
         print("Flooding mempool with many small claims")
         pegtxs = []
         sidechain.generate(101)
 
+        # Do mixture of raw peg-in and automatic peg-in tx construction
+        # where raw creation is done on another node
         for i in range(n_claims):
             addrs = sidechain.getpeginaddress()
             txid = parent.sendtoaddress(addrs["mainchain_address"], 1)
-            parent.generate(12)
+            parent.generate(1)
             proof = parent.gettxoutproof([txid])
             raw = parent.getrawtransaction(txid)
-            pegtxs += [sidechain.claimpegin(raw, proof)]
+            if i % 2 == 0:
+                parent.generate(11)
+                pegtxs += [sidechain.claimpegin(raw, proof)]
+            else:
+                # The raw API doesn't check for the additional 2 confirmation buffer
+                # So we only get 10 confirms then send off. Miners will add to block anyways.
+
+                # Don't mature whole way yet to test signing immature peg-in input
+                parent.generate(8)
+                # Wallet in sidechain2 gets funds instead of first
```

Comment:
> # Wallet in sidechain2 gets funds instead of first

I don't understand that sentence.

Reply:
Ah you mean "instead of wallet in sidechain1"?

Reply:
the `gettransaction` call later would fail if we don't select the right node since it's not in their wallet.

The "peg-in input" is not "in the wallet" at all. This is something we should probably improve at some point, but might require deeper changes.

Reply:
yes exactly, sorry